Add Clone trait to Unique

gilhooleyd opened the pull request:
Using a Unique iterator requires the Clone trait, so require this trait when creating the Unique object.

See the discussion in https://github.com/rust-itertools/itertools/pull/776

Philippe-Cholet commented:
I was not gonna suggest to make a test for it but I see your point.
I would only suggest to keep things as simple as possible with just `#[...] struct Item(u32);`.
And format to pass CI.

Philippe-Cholet commented:
Is the `unique` test not enough?!

gilhooleyd commented:
Thanks for both of your help! I think I've removed the test with the new CL.

I added the test originally when I was trying to remove `Clone`, and the struct I created didn't implement Clone.
You're right that now that Clone is required, this does the exact same thing as the existing tests.

I wish this functionality didn't require Clone, but I see now that it's necessary to do the in-place iteration while removing dups. Thank you again for your time!

jswrenn commented:
Ah, wait, could you also add it to the `struct` definition?

jswrenn commented:
Like so:
```rust
#[derive(Clone)]
#[must_use = "iterator adaptors are lazy and do nothing unless consumed"]
pub struct Unique<I: Iterator>
where
    I: Iterator,
    I::Item: Eq + Hash + Clone,
{
    iter: UniqueBy<I, I::Item, ()>,
}
```

gilhooleyd commented:
Done! Apologies for the churn

Philippe-Cholet commented:
It's not a problem but could you squash these commits into one simple commit?
